pkg/controller/deployment/sync.go

```diff
@@ -315,7 +315,7 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 
 	// new ReplicaSet does not exist, create one.
 	namespace := deployment.Namespace
-	podTemplateSpecHash := fmt.Sprintf("%d", deploymentutil.GetPodTemplateSpecHash(deployment.Spec.Template))
+	podTemplateSpecHash := fmt.Sprintf("%d", deploymentutil.GetPodTemplateSpecHashFnv(deployment.Spec.Template))
```

Comment:
We can't make this change for Deployments without providing a way to migrate old replica sets. I have a WIP open for managing hash migration for Deployments: https://github.com/kubernetes/kubernetes/pull/38714